---
title: "BCL to FastQ"
author: "Jared Crain"
date: "`r format(Sys.time(), '%B %d, %Y')`"
output: html_document
editor_options: 
  chunk_output_type: console
---

# Objective

Using blc-convert, set up sample scripts for each data source. Will develop shell scripts for demultiplexing and sample evaluation.

```{r setup, eval=TRUE}
knitr::opts_knit$set(root.dir = '/Users/crain/Documents/Bioinformatics/Demultiplex/') #can set paths but should be set by using Rproj
setwd('/Users/crain/Documents/Bioinformatics/Demultiplex') #set to path
options(digits = 10) #set options to include lots of digits

#Git link part of 
#https://github.com/jcrain1/Crain_Bioinformatics.git

#source('./scripts/Function_R.r') #load functions
```

# NEXUT0001

For NEXUT001, two lanes of NovaSeqX+ were ordered. Data was delivered as single lanes. Set up a demultiplex script for each one and then concatenate samples.

#TODO as original data came from Email [NEXUT0001.txt](./data/Original_Data/NEXUT0001.txt)

```{r process_NEXUT001, eval=FALSE}
require(RMySQL) #load required packages may need to install.packages("RMySQL") dependencies are DBI package as well

#from https://stackoverflow.com/questions/29809036/how-to-hide-asterisk-a-part-of-the-code-in-r get username and password non-human readable (caution this does not encrypt or hide password)
#This also only works in RStudio-may not work on all systems
username <- charToRaw(invisible(rstudioapi::askForPassword("Database username"))) #get username
password <-charToRaw(invisible(rstudioapi::askForPassword("Database password"))) #get password
iwg <- dbConnect(MySQL(),user = rawToChar(username), dbname = 'intermediate_wheatgrass', host = '127.0.0.1', password = rawToChar(password), port = 6306) #run this line to connect to the database

#Make a MySQL connection to beocat and intermediate_wheatgrass database
#iwg <- dbConnect(MySQL(),user = rawToChar(username), dbname = 'intermediate_wheatgrass', host = 'headnode.beocat.ksu.edu', password = rawToChar(password), port = 6306) #run this line to connect to the database  with VPN
#iwg <- dbConnect(MySQL(),user = rawToChar(username), dbname = 'intermediate_wheatgrass', host = '127.0.0.1', password = rawToChar(password), port = 6306) #run this line to connect to the database wihthout VPN after using ssh -L 6306:apate.beocat.ksu.edu:3306 jcrain@headnode.beocat.ksu.edu


#make key file
#added reverse compliment from database
nexut0001 <- "SELECT
nextera_iwg.nextera_id,
nextera_iwg.nextera_name,
dna_iwg.sample_id,
dna_iwg.sample_name,
nextera_i7barcode_iwg.barcode as 'i7_barcode',
nextera_i5barcode_iwg.barcode as 'i5_barcode',
nextera_i5barcode_iwg.barcode_reverse_comp as 'i5_reverse'
FROM dna_iwg LEFT JOIN nextera_iwg ON nextera_iwg.dna_id = dna_iwg.plate_id
JOIN nextera_i7barcode_iwg ON dna_iwg.well_A01 = nextera_i7barcode_iwg.well_A01
JOIN nextera_i5barcode_iwg ON nextera_iwg.i5 = nextera_i5barcode_iwg.ID
WHERE nextera_iwg.nextera_id like '%0030%'
ORDER BY nextera_iwg.nextera_id, dna_iwg.well_01A ASC"



dbDisconnect(iwg)

#write file
# write.table(nexut0001, file = './data/Original_Data/NEXUT0001.txt', quote = FALSE, row.names = FALSE, sep = '\t') #write file out


#clean up
rm(nexut0001, password, username)

```

Make sample files

```{r NEXUT0001_SampleFiles}
nexut0001 <- read.delim(file = './data/Original_Data/NEXUT0001.txt')

#check the number of unique reads
length(unique(nexut0001$sample_name))






```
From Laxman, need to remove header and only include plant_id and i7_i5 reverse complement to work with the [demultiplexNextera.pl](./scripts/demultiplexNextera.pl).

##Updated
Originally database did not have the i5 reverse complement, and the [demultiplexNextera.pl](./scripts/demultiplexNextera.pl) script had to have an exact match of the I1 and I2 index files. Following code is no longer needed and only kept as an example to get the reverse complement and expand an 8 base pair barcode to the full 10 base pairs that are sequenced in index run on NovaSeq.

```{r deprecated_complement_files}
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("Biostrings")

require(Biostrings) #for reverse complement

ffar30 <- read.table(file = './data/Original_Data/FFAR_Key_30.txt', header = TRUE, stringsAsFactors = FALSE)

ffar30$sample_name <- paste(ffar30$sample_name, ffar30$sample_id, ffar30$i5_barcode ,sep = '_')

#ADD 10 bp barcode example only
#ffar30$i7_barcode <- paste(ffar30$i7_barcode, 'AT', sep = '') #remove these as they add extra length depending on sequencing machine, demultiplex file is fixed now
#ffar30$i5_barcode <- paste('AC', ffar30$i5_barcode,  sep = '')

#add reverse complement example only
for(i in 1:nrow(ffar30)){
  ffar30$i5_rev[i] <- as.character(Biostrings::reverseComplement(DNAString(ffar30$i5_barcode[i])))
}

all(ffar30$i5_reverse == ffar30$i5_rev) #check database with computed reverse compliments

rm(ffar30, i)
```

# Make Key Files

Modify key files so that they are formatted to the [demultiplexNextera.pl](./scripts/demultiplexNextera.pl) specifications.

```{r key_files}
ffar30 <- read.table(file = './data/Original_Data/FFAR_Key_30.txt', header = TRUE, stringsAsFactors = FALSE)

ffar30$sample_name <- paste(ffar30$sample_name, ffar30$sample_id, ffar30$i5_barcode ,sep = '_')

ffar30$barcode <- paste(ffar30$i7_barcode, '_', ffar30$i5_reverse, sep = '')
ffar30 <- ffar30[, c(4, 8)]

length(unique(ffar30$sample_name)) == nrow(ffar30)

#write.table(ffar30, file = './data/Intermediate_File/FFAR_30_Key.txt', sep = '\t', col.names = FALSE, quote = FALSE, row.names = FALSE)


ffar32 <- read.table(file = './data/Original_Data/FFAR_Key_32.txt', header = TRUE, stringsAsFactors = FALSE)

ffar32$sample_name <- paste(ffar32$sample_name, ffar32$sample_id, ffar32$i5_barcode ,sep = '_')


ffar32$barcode <- paste(ffar32$i7_barcode, '_', ffar32$i5_reverse, sep = '')
ffar32 <- ffar32[, c(4, 8)]

length(unique(ffar32$sample_name)) == nrow(ffar32)

#write.table(ffar32, file = './data/Intermediate_File/FFAR_32_Key.txt', sep = '\t', col.names = FALSE, quote = FALSE, row.names = FALSE)

ffar33 <- read.table(file = './data/Original_Data/FFAR_Key_33.txt', header = TRUE, stringsAsFactors = FALSE)
ffar33$sample_name <- paste(ffar33$sample_name, ffar33$sample_id, ffar33$i5_barcode ,sep = '_')

ffar33$barcode <- paste(ffar33$i7_barcode, '_', ffar33$i5_reverse, sep = '')
ffar33 <- ffar33[, c(4, 8)]
length(unique(ffar33$sample_name)) == nrow(ffar33)


#write.table(ffar33, file = './data/Intermediate_File/FFAR_33_Key.txt', sep = '\t', col.names = FALSE, quote = FALSE, row.names = FALSE)

ffar34 <- read.table(file = './data/Original_Data/FFAR_Key_34.txt', header = TRUE, stringsAsFactors = FALSE)

ffar34$sample_name <- paste(ffar34$sample_name, ffar34$sample_id, ffar34$i5_barcode ,sep = '_')


ffar34$barcode <- paste(ffar34$i7_barcode, '_', ffar34$i5_reverse, sep = '')
ffar34 <- ffar34[, c(4, 8)]
length(unique(ffar34$sample_name)) == nrow(ffar34)


#write.table(ffar34, file = './data/Intermediate_File/FFAR_34_Key.txt', sep = '\t', col.names = FALSE, quote = FALSE, row.names = FALSE)

ffar35 <- read.table(file = './data/Original_Data/FFAR_Key_35.txt', header = TRUE, stringsAsFactors = FALSE)

ffar35$sample_name <- paste(ffar35$sample_name, ffar35$sample_id, ffar35$i5_barcode ,sep = '_')


ffar35$barcode <- paste(ffar35$i7_barcode, '_', ffar35$i5_reverse, sep = '')
ffar35 <- ffar35[, c(4, 8)]
length(unique(ffar35$sample_name)) == nrow(ffar35)

#write.table(ffar35, file = './data/Intermediate_File/FFAR_35_Key.txt', sep = '\t', col.names = FALSE, quote = FALSE, row.names = FALSE)

#clean up
rm(ffar30, ffar32, ffar33, ffar34, ffar35)
```

# Documentation of Beocat work:

After raw_data fastq files have been demultiplexed and concatenated into individual samples. The following directory structure and commands were used:

% mkdir 00_demultiplexed 01_samples 02_trimmed 03_aligned 04_filtered 05_indexed 06_vcf out_error

From running [demultiplexNextera.pl](./scripts/demultiplexNextera.pl) move all fastq files to 00_demultiplexed:
% mv *.fq.gz 00_demultiplexed

The original samples were moved to 00_demultiplexed and then concatenated together in 01_samples.

This preprocessing, resulted in paired end R1 and R2 files for each sample.

In the out_error directory, make directories to hold all of the output log files.
% cd out_error
% mkdir 00_demultiplexed 01_samples 02_trimmed 03_aligned 04_filtered 05_indexed 06_vcf

# Demultiplex 
For each Nextera library a script was set up to demultiplex the libraries. [00_FFAR30.sh](./scripts/00_FFAR30.sh), [00_FFAR32.sh](./scripts/00_FFAR32.sh), [00_FFAR33.sh](./scripts/00_FFAR33.sh), [00_FFAR34.sh](./scripts/00_FFAR34.sh), [00_FFAR35.sh](./scripts/00_FFAR35.sh). Scripts require little memory 1GB, but can take up to 20 days to complete. After scripts were completed all fastq files were moved to 00_demultiplexed.

% mv *.fq.gz /bulk/jpoland/skimseq/NEX00{00}/00_demultiplexed

% mv *.out *.err *.txt.txt out_error/00_demultiplexed

# Concatenate multiple same samples (different barcodes and wells into single sample)

```{r script_concat}
require(stringr)

#NEX0030
key30 <- read.delim(file = './data/Intermediate_File/FFAR_30_Key.txt', header = FALSE) #load key

colnames(key30) <- c('unique_id', 'barcode')

#do a gsub for blank dna to get BLANKDNA so we can split string
key30$sample <- key30$unique_id
key30$sample <- gsub('ank_', 'ank', key30$sample, ignore.case = TRUE) #gsub

#add well number with a substitute will be in blank wells first underscore
key30$sample <- ifelse(grepl('blank', key30$sample, ignore.case = TRUE), sub('_', '', key30$sample), key30$sample)

key30$main_sample <- sapply( str_split(key30$sample, pattern = '_'), '[[', 1)

#for each unique sample need to concatenate original files with new sample name for both R1 and R2 files
samples <- as.data.frame(unique(key30$main_sample))
colnames(samples) <- 'sample_name'

r1 <- cbind.data.frame(samples,  'R1') #make an R1 file
r2 <- cbind.data.frame(samples, 'R2') #make an R2 file


for(i in 1:nrow(r1)){#lloop over file and get all unique sample_id_barcodes
  #filenames start with these value so can cat start*R1* start*R1* > sample_name_R1.fq.gz
  x <- key30[key30$main_sample ==  r1$sample_name[i],]
  x$unique_id <- paste('./00_demultiplexed/', x$unique_id, '_R1.fq.gz', sep = '')
  files_to_cat <- paste( x$unique_id, collapse = ' ')
  r1$command[i] <- paste('cat ', files_to_cat, ' > ', paste('./01_samples/', toupper(r1$sample_name[i]), '_R1.fq.gz', sep = ''), sep = '' )
  }

#For R2 reads
for(i in 1:nrow(r2)){#lloop over file and get all unique sample_id_barcodes
  #filenames start with these value so can cat start*R2* start*R2* > sample_name_R2.fq.gz
  x <- key30[key30$main_sample == r2$sample_name[i],]
  x$unique_id <- paste('./00_demultiplexed/', x$unique_id, '_R2.fq.gz', sep = '')
  files_to_cat <- paste(x$unique_id, collapse = ' ')
 
  r2$command[i] <- paste('cat ', files_to_cat, ' > ', paste('./01_samples/', toupper(r2$sample_name[i]), '_R2.fq.gz', sep = '' ), sep = '')
  
}

#write files out
#write.table(r1[, 3], file = './data/Intermediate_File/FFAR30_Concat_R1.txt', row.names = FALSE, col.names = FALSE, quote = FALSE)
#write.table(r2[, 3], file = './data/Intermediate_File/FFAR30_Concat_R2.txt', row.names = FALSE, col.names = FALSE, quote = FALSE)

#clean up
rm(key30, r1, r2, samples, x, files_to_cat, i)

require(stringr)

#NEX0032
key32 <- read.delim(file = './data/Intermediate_File/FFAR_32_Key.txt', header = FALSE) #load key

colnames(key32) <- c('unique_id', 'barcode')

#do a gsub for blank dna to get BLANKDNAX so we can split string
key32$sample <- key32$unique_id
key32$sample <- gsub('ank_', 'ank', key32$sample, ignore.case = TRUE) #gsub

#add well number with a substitute will be in blank wells first underscore
key32$sample <- ifelse(grepl('blank', key32$sample, ignore.case = TRUE), sub('_', '', key32$sample), key32$sample)

key32$main_sample <- sapply( str_split(key32$sample, pattern = '_'), '[[', 1)

#for each unique sample need to concatenate original files with new sample name for both R1 and R2 files
samples <- as.data.frame(unique(key32$main_sample))
colnames(samples) <- 'sample_name'

r1 <- cbind.data.frame(samples,  'R1') #make an R1 file
r2 <- cbind.data.frame(samples, 'R2') #make an R2 file


for(i in 1:nrow(r1)){#lloop over file and get all unique sample_id_barcodes
  #filenames start with these value so can cat start*R1* start*R1* > sample_name_R1.fq.gz
  x <- key32[key32$main_sample == r1$sample_name[i],]
   x$unique_id <- paste('./00_demultiplexed/', x$unique_id, '_R1.fq.gz', sep = '')
  files_to_cat <- paste( x$unique_id, collapse = ' ')
  r1$command[i] <- paste('cat ', files_to_cat, ' > ', paste('./01_samples/', toupper(r1$sample_name[i]), '_R1.fq.gz', sep = ''), sep = '' )
  }

#For R2 reads
for(i in 1:nrow(r2)){#lloop over file and get all unique sample_id_barcodes
  #filenames start with these value so can cat start*R2* start*R2* > sample_name_R2.fq.gz
  x <- key32[key32$main_sample == r2$sample_name[i],]
  x$unique_id <- paste('./00_demultiplexed/', x$unique_id, '_R2.fq.gz', sep = '')
  files_to_cat <- paste(x$unique_id, collapse = ' ')
 
  r2$command[i] <- paste('cat ', files_to_cat, ' > ', paste('./01_samples/', toupper(r2$sample_name[i]), '_R2.fq.gz', sep = '' ), sep = '')
}

#write files out
#write.table(r1[, 3], file = './data/Intermediate_File/FFAR32_Concat_R1.txt', row.names = FALSE, col.names = FALSE, quote = FALSE)
#write.table(r2[, 3], file = './data/Intermediate_File/FFAR32_Concat_R2.txt', row.names = FALSE, col.names = FALSE, quote = FALSE)

#clean up
rm(key32, r1, r2, samples, x, files_to_cat, i)

#NEX0033
key33 <- read.delim(file = './data/Intermediate_File/FFAR_33_Key.txt', header = FALSE) #load key

colnames(key33) <- c('unique_id', 'barcode')

#do a gsub for blank dna to get BLANKDNAX so we can split string
key33$sample <- key33$unique_id
key33$sample <- gsub('ank_', 'ank', key33$sample, ignore.case = TRUE) #gsub

#add well number with a substitute will be in blank wells first underscore
key33$sample <- ifelse(grepl('blank', key33$sample, ignore.case = TRUE), sub('_', '', key33$sample), key33$sample)

key33$main_sample <- sapply( str_split(key33$sample, pattern = '_'), '[[', 1)

#for each unique sample need to concatenate original files with new sample name for both R1 and R2 files
samples <- as.data.frame(unique(key33$main_sample))
colnames(samples) <- 'sample_name'

r1 <- cbind.data.frame(samples,  'R1') #make an R1 file
r2 <- cbind.data.frame(samples, 'R2') #make an R2 file


for(i in 1:nrow(r1)){#lloop over file and get all unique sample_id_barcodes
  #filenames start with these value so can cat start*R1* start*R1* > sample_name_R1.fq.gz
  x <- key33[key33$main_sample == r1$sample_name[i],]
  x$unique_id <- paste('./00_demultiplexed/', x$unique_id, '_R1.fq.gz', sep = '')
  files_to_cat <- paste( x$unique_id, collapse = ' ')
  r1$command[i] <- paste('cat ', files_to_cat, ' > ', paste('./01_samples/', toupper(r1$sample_name[i]), '_R1.fq.gz', sep = ''), sep = '' )
  }

#For R2 reads
for(i in 1:nrow(r2)){#lloop over file and get all unique sample_id_barcodes
  #filenames start with these value so can cat start*R2* start*R2* > sample_name_R2.fq.gz
  x <- key33[key33$main_sample == r2$sample_name[i],]
  x$unique_id <- paste('./00_demultiplexed/', x$unique_id, '_R2.fq.gz', sep = '')
  files_to_cat <- paste(x$unique_id, collapse = ' ')
 
  r2$command[i] <- paste('cat ', files_to_cat, ' > ', paste('./01_samples/', toupper(r2$sample_name[i]), '_R2.fq.gz', sep = '' ), sep = '')
  
}

#write files out
#write.table(r1[, 3], file = './data/Intermediate_File/FFAR33_Concat_R1.txt', row.names = FALSE, col.names = FALSE, quote = FALSE)
#write.table(r2[, 3], file = './data/Intermediate_File/FFAR33_Concat_R2.txt', row.names = FALSE, col.names = FALSE, quote = FALSE)

#clean up
rm(key33, r1, r2, samples, x, files_to_cat, i)

#NEX0034
key34 <- read.delim(file = './data/Intermediate_File/FFAR_34_Key.txt', header = FALSE) #load key

colnames(key34) <- c('unique_id', 'barcode')

#do a gsub for blank dna to get BLANKDNAX so we can split string
key34$sample <- key34$unique_id
key34$sample <- gsub('ank_', 'ank', key34$sample, ignore.case = TRUE) #gsub

#add well number with a substitute will be in blank wells first underscore
key34$sample <- ifelse(grepl('blank', key34$sample, ignore.case = TRUE), sub('_', '', key34$sample), key34$sample)

key34$main_sample <- sapply( str_split(key34$sample, pattern = '_'), '[[', 1)

#for each unique sample need to concatenate original files with new sample name for both R1 and R2 files
samples <- as.data.frame(unique(key34$main_sample))
colnames(samples) <- 'sample_name'

r1 <- cbind.data.frame(samples,  'R1') #make an R1 file
r2 <- cbind.data.frame(samples, 'R2') #make an R2 file


for(i in 1:nrow(r1)){#lloop over file and get all unique sample_id_barcodes
  #filenames start with these value so can cat start*R1* start*R1* > sample_name_R1.fq.gz
  x <- key34[key34$main_sample == r1$sample_name[i],]
   x$unique_id <- paste('./00_demultiplexed/', x$unique_id, '_R1.fq.gz', sep = '')
  files_to_cat <- paste( x$unique_id, collapse = ' ')
  r1$command[i] <- paste('cat ', files_to_cat, ' > ', paste('./01_samples/', toupper(r1$sample_name[i]), '_R1.fq.gz', sep = ''), sep = '' )
  }

#For R2 reads
for(i in 1:nrow(r2)){#lloop over file and get all unique sample_id_barcodes
  #filenames start with these value so can cat start*R2* start*R2* > sample_name_R2.fq.gz
  x <- key34[key34$main_sample == r2$sample_name[i],]
   x$unique_id <- paste('./00_demultiplexed/', x$unique_id, '_R2.fq.gz', sep = '')
  files_to_cat <- paste(x$unique_id, collapse = ' ')
 
  r2$command[i] <- paste('cat ', files_to_cat, ' > ', paste('./01_samples/', toupper(r2$sample_name[i]), '_R2.fq.gz', sep = '' ), sep = '')
  
}

#write files out
#write.table(r1[, 3], file = './data/Intermediate_File/FFAR34_Concat_R1.txt', row.names = FALSE, col.names = FALSE, quote = FALSE)
#write.table(r2[, 3], file = './data/Intermediate_File/FFAR34_Concat_R2.txt', row.names = FALSE, col.names = FALSE, quote = FALSE)

#clean up
rm(key34, r1, r2, samples, x, files_to_cat, i)

#NEX0035
key35 <- read.delim(file = './data/Intermediate_File/FFAR_35_Key.txt', header = FALSE) #load key

colnames(key35) <- c('unique_id', 'barcode')

#do a gsub for blank dna to get BLANKDNAX so we can split string
key35$sample <- key35$unique_id
key35$sample <- gsub('ank_', 'ank', key35$sample, ignore.case = TRUE) #gsub

#add well number with a substitute will be in blank wells first underscore
key35$sample <- ifelse(grepl('blank', key35$sample, ignore.case = TRUE), sub('_', '', key35$sample), key35$sample)

key35$main_sample <- sapply( str_split(key35$sample, pattern = '_'), '[[', 1)

#for each unique sample need to concatenate original files with new sample name for both R1 and R2 files
samples <- as.data.frame(unique(key35$main_sample))
colnames(samples) <- 'sample_name'

r1 <- cbind.data.frame(samples,  'R1') #make an R1 file
r2 <- cbind.data.frame(samples, 'R2') #make an R2 file

# depending on how many rows are in R1 and R2, these can be sped up slightly by converting loops to map functions.
for(i in 1:nrow(r1)){#lloop over file and get all unique sample_id_barcodes
  #filenames start with these value so can cat start*R1* start*R1* > sample_name_R1.fq.gz
  x <- key35[key35$main_sample == r1$sample_name[i],]
  x$unique_id <- paste('./00_demultiplexed/', x$unique_id, '_R1.fq.gz', sep = '')
  files_to_cat <- paste( x$unique_id, collapse = ' ')
  r1$command[i] <- paste('cat ', files_to_cat, ' > ', paste('./01_samples/', toupper(r1$sample_name[i]), '_R1.fq.gz', sep = ''), sep = '' )
  }

#For R2 reads
for(i in 1:nrow(r2)){#lloop over file and get all unique sample_id_barcodes
  #filenames start with these value so can cat start*R2* start*R2* > sample_name_R2.fq.gz
  x <- key35[key35$main_sample == r2$sample_name[i],]
  x$unique_id <- paste('./00_demultiplexed/', x$unique_id, '_R2.fq.gz', sep = '')
  files_to_cat <- paste(x$unique_id, collapse = ' ')
 
  r2$command[i] <- paste('cat ', files_to_cat, ' > ', paste('./01_samples/', toupper(r2$sample_name[i]), '_R2.fq.gz', sep = '' ), sep = '')
}

#write files out
#write.table(r1[, 3], file = './data/Intermediate_File/FFAR35_Concat_R1.txt', row.names = FALSE, col.names = FALSE, quote = FALSE)
#write.table(r2[, 3], file = './data/Intermediate_File/FFAR35_Concat_R2.txt', row.names = FALSE, col.names = FALSE, quote = FALSE)

#clean up
rm(key35, r1, r2, samples, x, files_to_cat, i)

```
 
For each NEX library an R1 and R2 concat file was made as [FFAR35_Concat_R1.txt](./scripts/FFAR35_Concat_R1.txt), [FFAR35_Concat_R2.txt](./scripts/FFAR35_Concat_R2.txt), [FFAR34_Concat_R1.txt](./scripts/FFAR34_Concat_R1.txt), [FFAR34_Concat_R2.txt](./scripts/FFAR34_Concat_R2.txt), [FFAR33_Concat_R1.txt](./scripts/FFAR33_Concat_R1.txt), [FFAR33_Concat_R2.txt](./scripts/FFAR33_Concat_R2.txt), [FFAR32_Concat_R1.txt](./scripts/FFAR32_Concat_R1.txt), [FFAR32_Concat_R2.txt](./scripts/FFAR32_Concat_R2.txt), and [FFAR30_Concat_R1.txt](./scripts/FFAR30_Concat_R1.txt), [FFAR30_Concat_R2.txt](./scripts/FFAR30_Concat_R2.txt).

For both the R1 and R2 files, a shell script was developed using a common header command line and then pasting column three into the shell script as [01_FFAR30_Concat_R1.sh](./scripts/01_FFAR30_Concat_R1.sh), [01_FFAR30_Concat_R2.sh](./scripts/01_FFAR30_Concat_R2.sh), 
[01_FFAR32_Concat_R1.sh](./scripts/01_FFAR32_Concat_R1.sh), [01_FFAR32_Concat_R2.sh](./scripts/01_FFAR32_Concat_R2.sh),
[01_FFAR33_Concat_R1.sh](./scripts/01_FFAR33_Concat_R1.sh), [01_FFAR33_Concat_R2.sh](./scripts/01_FFAR33_Concat_R2.sh),
[01_FFAR34_Concat_R1.sh](./scripts/01_FFAR34_Concat_R1.sh), [01_FFAR34_Concat_R2.sh](./scripts/01_FFAR34_Concat_R2.sh), and
[01_FFAR35_Concat_R1.sh](./scripts/01_FFAR35_Concat_R1.sh), [01_FFAR35_Concat_R2.sh](./scripts/01_FFAR35_Concat_R2.sh).

Shell scripts were submitted to beocat using:
% sbatch 01_FFAR30_Concat_R1.sh 
% sbatch 01_FFAR30_Concat_R2.sh

% sbatch 01_FFAR32_Concat_R1.sh 
% sbatch 01_FFAR32_Concat_R2.sh

% sbatch 01_FFAR33_Concat_R1.sh 
% sbatch 01_FFAR33_Concat_R2.sh

% sbatch 01_FFAR34_Concat_R1.sh 
% sbatch 01_FFAR34_Concat_R2.sh

% sbatch 01_FFAR35_Concat_R1.sh 
% sbatch 01_FFAR35_Concat_R2.sh

Move error and out messages to ./out_error/01_samples/:
% mv *.out ./out_error/01_samples/

# Session Information

```{r sessionInformation}
sessionInfo()
```