---
title: "BCL to FastQ 2023"
author: "Jared Crain"
date: "`r format(Sys.time(), '%B %d, %Y')`"
output: html_document
editor_options: 
  chunk_output_type: console
---

# Objective

Using bcl-convert, set up sample scripts for each data source. Will develop shell scripts for demultiplexing and sample evaluation.

```{r setup, eval=TRUE}
knitr::opts_knit$set(root.dir = '/Users/crain/Documents/Bioinformatics/2023/00_Demultiplex/') #can set paths but should be set by using Rproj
setwd('/Users/crain/Documents/Bioinformatics/2023/00_Demultiplex/') #set to path
options(digits = 10) #set options to include lots of digits

#Git link part of 
#https://github.com/jcrain1/Crain_Bioinformatics.git

#source('./scripts/Function_R.r') #load functions
```

# CIMMYT AWG 2023
GBS files delivered as bcl. Need to convert bcl to original R1 fastq read with one fastq file per lane.

The basic key file was [HJM5MCCX2_L001_i7i5.csv](./data/Intermediate_File/HJM5MCCX2_L001_i7i5.csv) set up for each GBS library GBS1960-1992 including:
[HJM5MCCX2_L001_i7i5.csv](./data/Intermediate_File/HJM5MCCX2_L001_i7i5.csv),
[HJM5MCCX2_L002_i7i5.csv](./data/Intermediate_File/HJM5MCCX2_L002_i7i5.csv),
[HJM5MCCX2_L003_i7i5.csv](./data/Intermediate_File/HJM5MCCX2_L003_i7i5.csv),
[HJM5MCCX2_L004_i7i5.csv](./data/Intermediate_File/HJM5MCCX2_L004_i7i5.csv),
[HJM5MCCX2_L005_i7i5.csv](./data/Intermediate_File/HJM5MCCX2_L005_i7i5.csv),
[HJM5MCCX2_L006_i7i5.csv](./data/Intermediate_File/HJM5MCCX2_L006_i7i5.csv),
[HJM5MCCX2_L007_i7i5.csv](./data/Intermediate_File/HJM5MCCX2_L007_i7i5.csv),
[HJM5MCCX2_L008_i7i5.csv](./data/Intermediate_File/HJM5MCCX2_L008_i7i5.csv),
[HJMGHCCX2_L001_i7i5.csv](./data/Intermediate_File/HJMGHCCX2_L001_i7i5.csv),
[HJMGHCCX2_L002_i7i5.csv](./data/Intermediate_File/HJMGHCCX2_L002_i7i5.csv),
[HJMGHCCX2_L003_i7i5.csv](./data/Intermediate_File/HJMGHCCX2_L003_i7i5.csv),
[HJMVLCCX2_L008_i7i5.csv](./data/Intermediate_File/HJMVLCCX2_L008_i7i5.csv),
[HJM5MCCX2_L007_i7i5.csv](./data/Intermediate_File/HJM5MCCX2_L007_i7i5.csv),
[HJM5MCCX2_L008_i7i5.csv](./data/Intermediate_File/HJM5MCCX2_L008_i7i5.csv),
[HJMMJCCX2_L001_i7i5.csv](./data/Intermediate_File/HJMMJCCX2_L001_i7i5.csv),
[HJMMJCCX2_L002_i7i5.csv](./data/Intermediate_File/HJMMJCCX2_L002_i7i5.csv),
[HJMMJCCX2_L003_i7i5.csv](./data/Intermediate_File/HJMMJCCX2_L003_i7i5.csv),
[HJMMJCCX2_L004_i7i5.csv](./data/Intermediate_File/HJMMJCCX2_L004_i7i5.csv),
[HJMMJCCX2_L005_i7i5.csv](./data/Intermediate_File/HJMMJCCX2_L005_i7i5.csv),
[HJMMJCCX2_L006_i7i5.csv](./data/Intermediate_File/HJMMJCCX2_L006_i7i5.csv),

Files were ran using [BCL_Convert.sh](./scripts/BCL_Convert.sh) as

bash --exclude=warlock09 BCL_Convert.sh {INPUTDIR} {OUTPUTDIR} {SAMPLESHEET}

After a fastq file was created, the database was updated, and the R1 file was placed in /bulk/jpoland/gbs.

# NEX0067
#TODO

For NEX0067 (FFAR Phenomics Year 1), one lane of NovaSeq was ordered. Data was delivered as single lane. 


```{r process_NEX0067, eval=FALSE}
require(RMySQL) #load required packages may need to install.packages("RMySQL") dependencies are DBI package as well

#from https://stackoverflow.com/questions/29809036/how-to-hide-asterisk-a-part-of-the-code-in-r get username and password non-human readable (caution this does not encrypt or hide password)
#This also only works in RStudio-may not work on all systems
username <- charToRaw(invisible(rstudioapi::askForPassword("Database username"))) #get username
password <-charToRaw(invisible(rstudioapi::askForPassword("Database password"))) #get password

iwg <- dbConnect(MySQL(),user = rawToChar(username), dbname = 'intermediate_wheatgrass', host = '127.0.0.1', password = rawToChar(password), port = 6306) #run this line to connect to the database

#Make a MySQL connection to beocat and intermediate_wheatgrass database
#iwg <- dbConnect(MySQL(),user = rawToChar(username), dbname = 'intermediate_wheatgrass', host = 'headnode.beocat.ksu.edu', password = rawToChar(password), port = 6306) #run this line to connect to the database  with VPN
#iwg <- dbConnect(MySQL(),user = rawToChar(username), dbname = 'intermediate_wheatgrass', host = '127.0.0.1', password = rawToChar(password), port = 6306) #run this line to connect to the database wihthout VPN after using ssh -L 6306:apate.beocat.ksu.edu:3306 jcrain@headnode.beocat.ksu.edu


#make key file
#added reverse compliment from database
nex0067 <- dbGetQuery(iwg, "SELECT
nextera_iwg.nextera_id,
nextera_iwg.nextera_name,
dna_iwg.sample_id,
dna_iwg.sample_name,
nextera_i7barcode_iwg.barcode as 'i7_barcode',
nextera_i5barcode_iwg.barcode as 'i5_barcode',
nextera_i5barcode_iwg.barcode_reverse_comp as 'i5_reverse'
FROM dna_iwg LEFT JOIN nextera_iwg ON nextera_iwg.dna_id = dna_iwg.plate_id
JOIN nextera_i7barcode_iwg ON dna_iwg.well_A01 = nextera_i7barcode_iwg.well_A01
JOIN nextera_i5barcode_iwg ON nextera_iwg.i5 = nextera_i5barcode_iwg.ID
WHERE nextera_iwg.nextera_id like '%0067%'
ORDER BY nextera_iwg.nextera_id, dna_iwg.well_01A ASC")



dbDisconnect(iwg)

#write file
#write.table(nex0067, file = './data/Original_Data/NEX0067.txt', quote = FALSE, row.names = FALSE, sep = '\t') #write file out


#clean up
rm(nex0067, password, username, iwg)

```

Make sample files

```{r NEX0067_SampleFiles}
key <- read.delim(file = './data/Original_Data/NEX0067.txt')

#check the number of unique reads
length(unique(key$sample_name)) 

#For Each lane  
key$Lane <- 2
key <- key[, c('Lane', 'sample_name', 'i7_barcode', 'i5_barcode')] #get information for bcl2fastq processing

colnames(key) <- c('Lane', 'Sample_ID', 'index', 'index2')

#write out version sample file. Update number of index reads and reads as necessary. 
system('echo "[Header]\nFileFormatVersion,2\n\n[Reads]\nRead1Cycles,151\nRead2Cycles,151\nIndex1Cycles,10\nIndex2Cycles,10\n\n[BCLConvert_Settings]\nCreateFastqForIndexReads,0\nOverrideCycles,Y151;I8N2;N2I8;Y151\n[BCLConvert_Data]" > ./data/Intermediate_File/NEX0071_L007_i7i5.csv')

write.table(key, file = './data/Intermediate_File/NEX0071_L007_i7i5.csv', row.names = FALSE, quote = FALSE, append = TRUE, sep = ',') #get a warning message that is okay

#clean up
rm(key)




```

Using [BCL_Convert.sh](./scripts/BCL_Convert.sh), a shell script was started as
sbatch --exclude=warlock09 ~/Crain_Bioinformatics/2023/00_Demultiplex/scripts/BCL_Convert.sh  /fastscratch/jcrain/AN00014386/mmfs1/lustre2/Analysis/Project/AN00014386/tar/20230524_223JMNLT3_3/ /fastscratch/jcrain/NEXUT071_7 ~/Crain_Bioinformatics/2023/00_Demultiplex/data/Intermediate_File/NEX0071_L007_i7i5.csv








# NEX0072

For NEX0072 (CIMMYT GBS/SkimSeq/DarTSeq), two lanes of NovaSeqX+ were ordered. Data was delivered as single lanes. Set up a demultiplex script for each one and then concatenate samples.


```{r process_NEX0072, eval=FALSE}
require(RMySQL) #load required packages may need to install.packages("RMySQL") dependencies are DBI package as well

#from https://stackoverflow.com/questions/29809036/how-to-hide-asterisk-a-part-of-the-code-in-r get username and password non-human readable (caution this does not encrypt or hide password)
#This also only works in RStudio-may not work on all systems
username <- charToRaw(invisible(rstudioapi::askForPassword("Database username"))) #get username
password <-charToRaw(invisible(rstudioapi::askForPassword("Database password"))) #get password

wheat <- dbConnect(MySQL(),user = rawToChar(username), dbname = 'wheatgenetics', host = '127.0.0.1', password = rawToChar(password), port = 6306) #run this line to connect to the database

#Make a MySQL connection to beocat and intermediate_wheatgrass database
#wheat <- dbConnect(MySQL(),user = rawToChar(username), dbname = 'wheatgenetics', host = 'headnode.beocat.ksu.edu', password = rawToChar(password), port = 6306) #run this line to connect to the database  with VPN
#wheat <- dbConnect(MySQL(),user = rawToChar(username), dbname = 'wheatgenetics', host = '127.0.0.1', password = rawToChar(password), port = 6306) #run this line to connect to the database wihthout VPN after using ssh -L 6306:apate.beocat.ksu.edu:3306 jcrain@headnode.beocat.ksu.edu


#make key file
#added reverse compliment from database
nex0072 <- dbGetQuery(wheat, "SELECT
nextera.nextera_id,
nextera.nextera_name,
dna.sample_id,
dna.sample_name,
nextera_i7barcode.barcode as 'i7_barcode',
nextera_i5barcode.barcode as 'i5_barcode',
nextera_i5barcode.barcode_reverse_comp as 'i5_reverse'
FROM dna LEFT JOIN nextera ON nextera.dna_id = dna.plate_id
JOIN nextera_i7barcode ON dna.well_A01 = nextera_i7barcode.well_A01
JOIN nextera_i5barcode ON nextera.i5 = nextera_i5barcode.ID
WHERE nextera.nextera_id like '%0072%'
ORDER BY nextera.nextera_id, dna.well_01A ASC")

dbDisconnect(wheat)

#write file
#write.table(nex0072, file = './data/Original_Data/NEX0072.txt', quote = FALSE, row.names = FALSE, sep = '\t') #write file out

#clean up
rm(nex0072, password, username, wheat)

```

Make sample files

```{r NEX0072_SampleFiles}
key <- read.delim(file = './data/Original_Data/NEX0072.txt')

#check the number of unique reads
length(unique(key$sample_name)) 

#For Each lane 
key$Lane <- 8
key <- key[, c('Lane', 'sample_name', 'i7_barcode', 'i5_barcode')] #get information for bcl2fastq processing

colnames(key) <- c('Lane', 'Sample_ID', 'index', 'index2')

#write out version 2 sample file. Update number of index reads and reads as necessary. 
system('echo "[Header]\nFileFormatVersion,2\n\n[Reads]\nRead1Cycles,151\nRead2Cycles,151\nIndex1Cycles,10\nIndex2Cycles,10\n\n[BCLConvert_Settings]\nCreateFastqForIndexReads,0\nOverrideCycles,Y151;I8N2;N2I8;Y151\n[BCLConvert_Data]" > ./data/Intermediate_File/NEX0072_L008_i7i5.csv')

write.table(key, file = './data/Intermediate_File/NEX0072_L008_i7i5.csv', row.names = FALSE, quote = FALSE, append = TRUE, sep = ',') #get a warning message that is okay


#For Each lane 
key$Lane <- 3

#write out version 2 sample file. Update number of index reads and reads as necessary. 
system('echo "[Header]\nFileFormatVersion,2\n\n[Reads]\nRead1Cycles,151\nRead2Cycles,151\nIndex1Cycles,10\nIndex2Cycles,10\n\n[BCLConvert_Settings]\nCreateFastqForIndexReads,0\nOverrideCycles,Y151;I8N2;N2I8;Y151\n[BCLConvert_Data]" > ./data/Intermediate_File/NEX0072_L003_i7i5.csv')

write.table(key, file = './data/Intermediate_File/NEX0072_L003_i7i5.csv', row.names = FALSE, quote = FALSE, append = TRUE, sep = ',') #get a warning message that is okay

#clean up
rm(key)
```

Using [BCL_Convert.sh](./scripts/BCL_Convert.sh), a shell script was started as
sbatch --exclude=warlock09 ~/Crain_Bioinformatics/2023/00_Demultiplex/scripts/BCL_Convert.sh  /fastscratch/jcrain/AN00014387/mmfs1/lustre2/Analysis/Project/AN00014387/tar/20230524_223JMNLT3_3/ /fastscratch/jcrain/NEXUT071_7 ~/Crain_Bioinformatics/2023/00_Demultiplex/data/Intermediate_File/NEX0072_L008_i7i5.csv


sbatch --exclude=warlock09 ~/Crain_Bioinformatics/2023/00_Demultiplex/scripts/BCL_Convert.sh  /fastscratch/jcrain/AN00014387_2ndLane/mmfs1/lustre2/Analysis/Project/AN00014387/tar/20230626_225N32LT3_5/ /fastscratch/jcrain/NEX0072_3 ~/Crain_Bioinformatics/2023/00_Demultiplex/data/Intermediate_File/NEX0072_L003_i7i5.csv







# NEX0071

For NEX0071 (FFAR Phenomics Year 2), two lanes of NovaSeqX+ were ordered. Data was delivered as single lanes. Set up a demultiplex script for each one and then concatenate samples.


```{r process_NEX0071, eval=FALSE}
require(RMySQL) #load required packages may need to install.packages("RMySQL") dependencies are DBI package as well

#from https://stackoverflow.com/questions/29809036/how-to-hide-asterisk-a-part-of-the-code-in-r get username and password non-human readable (caution this does not encrypt or hide password)
#This also only works in RStudio-may not work on all systems
username <- charToRaw(invisible(rstudioapi::askForPassword("Database username"))) #get username
password <-charToRaw(invisible(rstudioapi::askForPassword("Database password"))) #get password

iwg <- dbConnect(MySQL(),user = rawToChar(username), dbname = 'intermediate_wheatgrass', host = '127.0.0.1', password = rawToChar(password), port = 6306) #run this line to connect to the database

#Make a MySQL connection to beocat and intermediate_wheatgrass database
#iwg <- dbConnect(MySQL(),user = rawToChar(username), dbname = 'intermediate_wheatgrass', host = 'headnode.beocat.ksu.edu', password = rawToChar(password), port = 6306) #run this line to connect to the database  with VPN
#iwg <- dbConnect(MySQL(),user = rawToChar(username), dbname = 'intermediate_wheatgrass', host = '127.0.0.1', password = rawToChar(password), port = 6306) #run this line to connect to the database wihthout VPN after using ssh -L 6306:apate.beocat.ksu.edu:3306 jcrain@headnode.beocat.ksu.edu


#make key file
#added reverse compliment from database
nex0071 <- dbGetQuery(iwg, "SELECT
nextera_iwg.nextera_id,
nextera_iwg.nextera_name,
dna_iwg.sample_id,
dna_iwg.sample_name,
nextera_i7barcode_iwg.barcode as 'i7_barcode',
nextera_i5barcode_iwg.barcode as 'i5_barcode',
nextera_i5barcode_iwg.barcode_reverse_comp as 'i5_reverse'
FROM dna_iwg LEFT JOIN nextera_iwg ON nextera_iwg.dna_id = dna_iwg.plate_id
JOIN nextera_i7barcode_iwg ON dna_iwg.well_A01 = nextera_i7barcode_iwg.well_A01
JOIN nextera_i5barcode_iwg ON nextera_iwg.i5 = nextera_i5barcode_iwg.ID
WHERE nextera_iwg.nextera_id like '%0071%'
ORDER BY nextera_iwg.nextera_id, dna_iwg.well_01A ASC")



dbDisconnect(iwg)

#write file
#write.table(nex0071, file = './data/Original_Data/NEX0071.txt', quote = FALSE, row.names = FALSE, sep = '\t') #write file out


#clean up
rm(nex0071, password, username, iwg)

```

Make sample files

```{r NEX0071_SampleFiles}
key <- read.delim(file = './data/Original_Data/NEX0071.txt')

#check the number of unique reads
length(unique(key$sample_name)) #17 duplicated samples

#For Each lane 2 
key$Lane <- 7
key <- key[, c('Lane', 'sample_name', 'i7_barcode', 'i5_barcode')] #get information for bcl2fastq processing

colnames(key) <- c('Lane', 'Sample_ID', 'index', 'index2')

#write out version 2 sample file. Update number of index reads and reads as necessary. 
system('echo "[Header]\nFileFormatVersion,2\n\n[Reads]\nRead1Cycles,151\nRead2Cycles,151\nIndex1Cycles,10\nIndex2Cycles,10\n\n[BCLConvert_Settings]\nCreateFastqForIndexReads,0\nOverrideCycles,Y151;I8N2;N2I8;Y151\n[BCLConvert_Data]" > ./data/Intermediate_File/NEX0071_L007_i7i5.csv')

write.table(key, file = './data/Intermediate_File/NEX0071_L007_i7i5.csv', row.names = FALSE, quote = FALSE, append = TRUE, sep = ',') #get a warning message that is okay


#For Each lane 
key$Lane <- 2

#write out version 2 sample file. Update number of index reads and reads as necessary. 
system('echo "[Header]\nFileFormatVersion,2\n\n[Reads]\nRead1Cycles,151\nRead2Cycles,151\nIndex1Cycles,10\nIndex2Cycles,10\n\n[BCLConvert_Settings]\nCreateFastqForIndexReads,0\nOverrideCycles,Y151;I8N2;N2I8;Y151\n[BCLConvert_Data]" > ./data/Intermediate_File/NEX0071_L002_i7i5.csv')

write.table(key, file = './data/Intermediate_File/NEX0071_L002_i7i5.csv', row.names = FALSE, quote = FALSE, append = TRUE, sep = ',') #get a warning message that is okay

#clean up
rm(key)




```

Using [BCL_Convert.sh](./scripts/BCL_Convert.sh), a shell script was started as
sbatch --exclude=warlock09 ~/Crain_Bioinformatics/2023/00_Demultiplex/scripts/BCL_Convert.sh  /fastscratch/jcrain/AN00014386/mmfs1/lustre2/Analysis/Project/AN00014386/tar/20230524_223JMNLT3_3/ /fastscratch/jcrain/NEXUT071_7 ~/Crain_Bioinformatics/2023/00_Demultiplex/data/Intermediate_File/NEX0071_L007_i7i5.csv


sbatch --exclude=warlock09 ~/Crain_Bioinformatics/2023/00_Demultiplex/scripts/BCL_Convert.sh  /fastscratch/jcrain/AN00014386_2ndLane/mmfs1/lustre2/Analysis/Project/AN00014386/tar/20230626_225N32LT3_5/ /fastscratch/jcrain/NEX0071_2 ~/Crain_Bioinformatics/2023/00_Demultiplex/data/Intermediate_File/NEX0071_L002_i7i5.csv







# NEXUT0001

For NEXUT001, two lanes of NovaSeqX+ were ordered. Data was delivered as single lanes. Set up a demultiplex script for each one and then concatenate samples.

##TODO as original data came from Email [NEXUT0001.txt](./data/Original_Data/NEXUT0001.txt)

```{r process_NEXUT001, eval=FALSE}
require(RMySQL) #load required packages may need to install.packages("RMySQL") dependencies are DBI package as well

#from https://stackoverflow.com/questions/29809036/how-to-hide-asterisk-a-part-of-the-code-in-r get username and password non-human readable (caution this does not encrypt or hide password)
#This also only works in RStudio-may not work on all systems
username <- charToRaw(invisible(rstudioapi::askForPassword("Database username"))) #get username
password <-charToRaw(invisible(rstudioapi::askForPassword("Database password"))) #get password
iwg <- dbConnect(MySQL(),user = rawToChar(username), dbname = 'intermediate_wheatgrass', host = '127.0.0.1', password = rawToChar(password), port = 6306) #run this line to connect to the database

#Make a MySQL connection to beocat and intermediate_wheatgrass database
#iwg <- dbConnect(MySQL(),user = rawToChar(username), dbname = 'intermediate_wheatgrass', host = 'headnode.beocat.ksu.edu', password = rawToChar(password), port = 6306) #run this line to connect to the database  with VPN
#iwg <- dbConnect(MySQL(),user = rawToChar(username), dbname = 'intermediate_wheatgrass', host = '127.0.0.1', password = rawToChar(password), port = 6306) #run this line to connect to the database wihthout VPN after using ssh -L 6306:apate.beocat.ksu.edu:3306 jcrain@headnode.beocat.ksu.edu


#make key file
#added reverse compliment from database
nexut0001 <- "SELECT
nextera_iwg.nextera_id,
nextera_iwg.nextera_name,
dna_iwg.sample_id,
dna_iwg.sample_name,
nextera_i7barcode_iwg.barcode as 'i7_barcode',
nextera_i5barcode_iwg.barcode as 'i5_barcode',
nextera_i5barcode_iwg.barcode_reverse_comp as 'i5_reverse'
FROM dna_iwg LEFT JOIN nextera_iwg ON nextera_iwg.dna_id = dna_iwg.plate_id
JOIN nextera_i7barcode_iwg ON dna_iwg.well_A01 = nextera_i7barcode_iwg.well_A01
JOIN nextera_i5barcode_iwg ON nextera_iwg.i5 = nextera_i5barcode_iwg.ID
WHERE nextera_iwg.nextera_id like '%0030%'
ORDER BY nextera_iwg.nextera_id, dna_iwg.well_01A ASC"



dbDisconnect(iwg)

#write file
# write.table(nexut0001, file = './data/Original_Data/NEXUT0001.txt', quote = FALSE, row.names = FALSE, sep = '\t') #write file out


#clean up
rm(nexut0001, password, username)

```

Make sample files

```{r NEXUT0001_SampleFiles}
key <- read.delim(file = './data/Original_Data/NEXUT0001.txt')

#check the number of unique reads
length(unique(key$sample_name))

#For Each lane 2 
key$Lane <- 2
key <- key[, c('Lane', 'sample_name', 'i7_barcode', 'i5_barcode')] #get information for bcl2fastq processing

colnames(key) <- c('Lane', 'Sample_ID', 'index', 'index2')

#write out version 2 sample file. Update number of index reads and reads as necessary. 
system('echo "[Header]\nFileFormatVersion,2\n\n[Reads]\nRead1Cycles,151\nRead2Cycles,151\nIndex1Cycles,10\nIndex2Cycles,10\n\n[BCLConvert_Settings]\nCreateFastqForIndexReads,0\nOverrideCycles,Y151;I8N2;N2I8;Y151\n[BCLConvert_Data]" > ./data/Intermediate_File/NEXUT0001_L002_i7i5.csv')

write.table(key, file = './data/Intermediate_File/NEXUT0001_L002_i7i5.csv', row.names = FALSE, quote = FALSE, append = TRUE, sep = ',') #get a warning message that is okay


#For Each lane 8
key$Lane <- 8

#write out version 2 sample file. Update number of index reads and reads as necessary. 
system('echo "[Header]\nFileFormatVersion,2\n\n[Reads]\nRead1Cycles,151\nRead2Cycles,151\nIndex1Cycles,10\nIndex2Cycles,10\n\n[BCLConvert_Settings]\nCreateFastqForIndexReads,0\nOverrideCycles,Y151;I8N2;N2I8;Y151\n[BCLConvert_Data]" > ./data/Intermediate_File/NEXUT0001_L008_i7i5.csv')

write.table(key, file = './data/Intermediate_File/NEXUT0001_L008_i7i5.csv', row.names = FALSE, quote = FALSE, append = TRUE, sep = ',') #get a warning message that is okay

#clean up
rm(key)




```

Using [BCL_Convert.sh](./scripts/BCL_Convert.sh), a shell script was started as
sbatch --exclude=warlock09 ~/Crain_Bioinformatics/2023/00_Demultiplex/scripts/BCL_Convert.sh  /fastscratch/jcrain/AN00014624/mmfs1/lustre2/Analysis/Project/AN00014624/tar/20230612_225MNMLT3_6/ /fastscratch/jcrain/NEXUT001_2 ~/Crain_Bioinformatics/2023/00_Demultiplex/data/Intermediate_File/NEXUT0001_L002_i7i5.csv


sbatch --exclude=warlock09 ~/Crain_Bioinformatics/2023/00_Demultiplex/scripts/BCL_Convert.sh  /fastscratch/jcrain/AN00014624_2ndLane/mmfs1/lustre2/Analysis/Project/AN00014624/tar/20230626_225N32LT3_5/ /fastscratch/jcrain/NEXUT001_8 ~/Crain_Bioinformatics/2023/00_Demultiplex/data/Intermediate_File/NEXUT0001_L008_i7i5.csv






# NEXUT0002

For NEXUT002, two lanes of NovaSeqX+ were ordered. Data was delivered on same flowcell. Set up a demultiplex script that will combine lanes.

##TODO as original data came from Email [Skim_Seq_NXUT0002.txt](./data/Original_Data/Skim_Seq_NXUT0002.txt)

```{r process_NEXUT002, eval=FALSE}
require(RMySQL) #load required packages may need to install.packages("RMySQL") dependencies are DBI package as well

#from https://stackoverflow.com/questions/29809036/how-to-hide-asterisk-a-part-of-the-code-in-r get username and password non-human readable (caution this does not encrypt or hide password)
#This also only works in RStudio-may not work on all systems
username <- charToRaw(invisible(rstudioapi::askForPassword("Database username"))) #get username
password <-charToRaw(invisible(rstudioapi::askForPassword("Database password"))) #get password
iwg <- dbConnect(MySQL(),user = rawToChar(username), dbname = 'intermediate_wheatgrass', host = '127.0.0.1', password = rawToChar(password), port = 6306) #run this line to connect to the database

#Make a MySQL connection to beocat and intermediate_wheatgrass database
#iwg <- dbConnect(MySQL(),user = rawToChar(username), dbname = 'intermediate_wheatgrass', host = 'headnode.beocat.ksu.edu', password = rawToChar(password), port = 6306) #run this line to connect to the database  with VPN
#iwg <- dbConnect(MySQL(),user = rawToChar(username), dbname = 'intermediate_wheatgrass', host = '127.0.0.1', password = rawToChar(password), port = 6306) #run this line to connect to the database wihthout VPN after using ssh -L 6306:apate.beocat.ksu.edu:3306 jcrain@headnode.beocat.ksu.edu


#make key file
#added reverse compliment from database
nexut0002 <- "SELECT
nextera_iwg.nextera_id,
nextera_iwg.nextera_name,
dna_iwg.sample_id,
dna_iwg.sample_name,
nextera_i7barcode_iwg.barcode as 'i7_barcode',
nextera_i5barcode_iwg.barcode as 'i5_barcode',
nextera_i5barcode_iwg.barcode_reverse_comp as 'i5_reverse'
FROM dna_iwg LEFT JOIN nextera_iwg ON nextera_iwg.dna_id = dna_iwg.plate_id
JOIN nextera_i7barcode_iwg ON dna_iwg.well_A01 = nextera_i7barcode_iwg.well_A01
JOIN nextera_i5barcode_iwg ON nextera_iwg.i5 = nextera_i5barcode_iwg.ID
WHERE nextera_iwg.nextera_id like '%0030%'
ORDER BY nextera_iwg.nextera_id, dna_iwg.well_01A ASC"



dbDisconnect(iwg)

#write file
# write.table(nexut0002, file = './data/Original_Data/NEXUT0002.txt', quote = FALSE, row.names = FALSE, sep = '\t') #write file out


#clean up
rm(nexut0002, password, username)

```

Make sample files

```{r NEXUT0002_SampleFiles}
key <- read.delim(file = './data/Original_Data/Skim_Seq_NXUT0002.txt')

#check the number of unique reads
length(unique(key$sample_name))

#For Each lane 2 
key$Lane <- 5
key <- key[, c('Lane', 'sample_name', 'i7_barcode', 'i5_barcode')] #get information for bcl2fastq processing
key2 <- key
key2$Lane <- 6
key <- rbind(key, key2)

colnames(key) <- c('Lane', 'Sample_ID', 'index', 'index2')

#write out version 2 sample file. Update number of index reads and reads as necessary. 
#system('echo "[Header]\nFileFormatVersion,2\n\n[Reads]\nRead1Cycles,151\nRead2Cycles,151\nIndex1Cycles,10\nIndex2Cycles,10\n\n[BCLConvert_Settings]\nCreateFastqForIndexReads,0\nOverrideCycles,Y151;I8N2;N2I8;Y151\n[BCLConvert_Data]" > ./data/Intermediate_File/NEXUT0002_L005_6_i7i5.csv')

#write.table(key, file = './data/Intermediate_File/NEXUT0002_L005_6_i7i5.csv', row.names = FALSE, quote = FALSE, append = TRUE, sep = ',') #get a warning message that is okay

#clean up
rm(key)


```

Using [BCL_Convert.sh](./scripts/BCL_Convert.sh), a shell script was started as
sbatch --exclude=warlock09 ~/Crain_Bioinformatics/2023/00_Demultiplex/scripts/BCL_Convert.sh  /fastscratch/jcrain/AN00015244/mmfs1/lustre2/Analysis/Project/AN00015244/tar/20230725_2273TTLT3_5/ /fastscratch/jcrain/NEXUT002 ~/Crain_Bioinformatics/2023/00_Demultiplex/data/Intermediate_File/NEXUT0002_L005_6_i7i5.csv




# Session Information

```{r sessionInformation}
sessionInfo()
```